### PR TITLE
🌱[e2e] Install CNI using ClusterResourceSet for e2e tests

### PR DESCRIPTION
**Pull request by sedefsavas**
**What this PR does / why we need it**:
This PR installs e2e CNIs using ClusterResourceSet.

This PR should hold until CAPI changes are merged: https://github.com/kubernetes-sigs/cluster-api/pull/3332 and
CAPZ PR: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/785
/hold

Release note:

NONE

### CODE REVIEW

**Comment by sedefsavas** [comment]
cc @fabriziopandini

File: test/e2e/common.go [hunk 1]
```diff
@@ -143,3 +146,10 @@ func redactLogs() {
 	cmd := exec.Command(e2eConfig.GetVariable(RedactLogScriptPath))
 	cmd.Run()
 }
+
+func setCNIResources(cniManifestPath string) {
+	cniData, err := ioutil.ReadFile(cniManifestPath)
+	Expect(err).ToNot(HaveOccurred(), "Failed to read the e2e test CNI file")
+	Expect(cniData).ToNot(BeEmpty(), "CNI file should not be empty")
+	Expect(os.Setenv(CNIResources, base64.StdEncoding.EncodeToString(cniData)))
+}
```
Comment (CecileRobertMichon) [inline test/e2e/common.go:155]: missing newline

File: test/e2e/common.go [hunk 2]
```diff
@@ -39,6 +41,7 @@ import (
 const (
 	KubernetesVersion   = "KUBERNETES_VERSION"
 	CNIPath             = "CNI"
+	CNIResources                 = "CNI_RESOURCES"
```
Comment (CecileRobertMichon) [inline test/e2e/common.go:44]: extra indent?

File: templates/test/patches/cluster-resource-set.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cni-calico
+type: addons.cluster.x-k8s.io/resource-set
+data:
+  addon.yaml: ${ CNI_RESOURCES }
```
Comment (CecileRobertMichon) [inline templates/test/patches/cluster-resource-set.yaml:7]:
```suggestion
  addon.yaml: ${CNI_RESOURCES}
```

**Comment by sedefsavas** [comment]
> Does this require the upcoming CAPI release to work?

Yes, it may go in CAPI v0.3.7 release.

**Comment by CecileRobertMichon** [comment]
@sedefsavas please add a release note in the PR description

**Comment by CecileRobertMichon** [comment]
this also requires #787 to go in first, correct?

**Comment by sedefsavas** [comment]
@CecileRobertMichon Good point, don't know how e2e tests are implemented here but we should be able to enable experimental features by setting variables for clusterctl in azure-dev.yaml.
Trying to see if we need #787

**Comment by sedefsavas** [comment]
This PR requires #787 to get merged.

**Comment by CecileRobertMichon** [comment]
@sedefsavas is this ready to move forward now that #787 has merged?

**Comment by sedefsavas** [comment]
@CecileRobertMichon not yet.
All providers should merge it at the same time with CAPI changes https://github.com/kubernetes-sigs/cluster-api/pull/3332

**Comment by nader-ziada** [comment]
@sedefsavas created a new PR to replace this with the newer changes #918 (added you as a co-author since you started the work)

**Comment by sedefsavas** [comment]
Thanks @nader-ziada, closing this one.

**Pull request closed without merging**